Framework/DataHandling/src/LoadILLDiffraction.cpp

```diff
@@ -0,0 +1,446 @@
+  return "DataHandling\\Nexus;ILL\\Diffraction";
+}
+
+/// Algorithm's summary for use in the GUI and help. @see Algorithm::summary
+const std::string LoadILLDiffraction::summary() const {
+  return "Loads ILL diffraction nexus files.";
+}
+
+/**
+ * Constructor
+ */
+LoadILLDiffraction::LoadILLDiffraction()
+    : IFileLoader<NexusDescriptor>(), m_instNames({"D20"}) {}
+
+/**
+ * Initialize the algorithm's properties.
+ */
+void LoadILLDiffraction::init() {
+  declareProperty(
+      make_unique<FileProperty>("Filename", "", FileProperty::Load, ".nxs"),
+      "File path of the data file to load");
+  declareProperty(make_unique<WorkspaceProperty<MatrixWorkspace>>(
+                      "OutputWorkspace", "", Direction::Output),
+                  "The output workspace.");
+}
+
+/**
+ * Executes the algorithm.
+ */
+void LoadILLDiffraction::exec() {
+
+  Progress progress(this, 0, 1, 3);
+
+  m_fileName = getPropertyValue("Filename");
+
+  loadScanVars();
+  progress.report("Loaded the scanned variables");
+
+  loadDataScan();
+  progress.report("Loaded the detector scan data");
+
+  loadMetaData();
+  progress.report("Loaded the metadata");
+
+  setProperty("OutputWorkspace", m_outWorkspace);
+}
+
+/**
+* Loads the scanned detector data
+*/
+void LoadILLDiffraction::loadDataScan() {
+
+  // open the root entry
+  NXRoot dataRoot(m_fileName);
+  NXEntry firstEntry = dataRoot.openFirstEntry();
+
+  m_instName = firstEntry.getString("instrument/name");
+
+  m_startTime = DateAndTime(
+      m_loadHelper.dateTimeInIsoFormat(firstEntry.getString("start_time")));
+
+  // read the detector data
+  NXData dataGroup = firstEntry.openNXData("data_scan/detector_data");
+  NXUInt data = dataGroup.openUIntData();
+  data.load();
+
+  // read the scan data
+  NXData scanGroup = firstEntry.openNXData("data_scan/scanned_variables");
+  NXDouble scan = scanGroup.openDoubleData();
+  scan.load();
+
+  // read which variables are scanned
+  NXInt scanned = firstEntry.openNXInt(
+      "data_scan/scanned_variables/variables_names/scanned");
+  scanned.load();
+
+  // read what is going to be the axis
+  NXInt axis =
+      firstEntry.openNXInt("data_scan/scanned_variables/variables_names/axis");
+  axis.load();
+
+  // read the starting two theta
+  NXFloat twoTheta0 = firstEntry.openNXFloat("instrument/2theta/value");
+  twoTheta0.load();
+
+  // figure out the dimensions
+  m_numberDetectorsRead = static_cast<size_t>(data.dim1());
+  m_numberScanPoints = static_cast<size_t>(data.dim0());
+
+  // set which scanned variables are scanned, which should be the axis
+  for (size_t i = 0; i < m_scanVar.size(); ++i) {
+    m_scanVar[i].setAxis(axis[static_cast<int>(i)]);
+    m_scanVar[i].setScanned(scanned[static_cast<int>(i)]);
+  }
+
+  resolveScanType();
+
+  resolveInstrument();
+
+  initWorkspace();
+
+  fillDataScanMetaData(scan);
+
+  if (m_scanType == DetectorScan) {
+    fillMovingInstrumentScan(data, scan);
```

Comment:
This is a place holder that will be filled later on in a separate PR when the scanning workspace functionality is ready. I can remove it for now, but this part must be refactored later anyways.